{
    "comments": [
        {
            "author": null,
            "body": "Here's my stack trace that left me baffled at first:\n\njavax.servlet.ServletException: Exception processing JAR at resource path\n/WEB-INF/lib/log4j-1.2.8.jar\n\tat org.apache.catalina.startup.ContextConfig.tldScanJar(ContextConfig.java:930)\n\tat org.apache.catalina.startup.ContextConfig.tldScan(ContextConfig.java:868)\n\tat org.apache.catalina.startup.ContextConfig.start(ContextConfig.java:647)\n\tat org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:243)\n\tat\norg.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:166)\n\tat org.apache.catalina.core.StandardContext.start(StandardContext.java:3582)\n\tat org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1188)\n\tat org.apache.catalina.core.StandardHost.start(StandardHost.java:754)\n\tat org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1188)\n\tat org.apache.catalina.core.StandardEngine.start(StandardEngine.java:363)\n\tat org.apache.catalina.core.StandardService.start(StandardService.java:497)\n\tat org.apache.catalina.core.StandardServer.start(StandardServer.java:2190)\n\tat org.apache.catalina.startup.Catalina.start(Catalina.java:512)\n\tat org.apache.catalina.startup.Catalina.execute(Catalina.java:400)\n\tat org.apache.catalina.startup.Catalina.process(Catalina.java:180)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat\nsun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:324)\n\tat org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:203)\n----- Root Cause -----\njava.io.FileNotFoundException\n\tat\norg.apache.naming.resources.DirContextURLConnection.getInputStream(DirContextURLConnection.java:344)\n\tat sun.net.www.protocol.jar.URLJarFile.retrieve(URLJarFile.java:161)\n\tat sun.net.www.protocol.jar.URLJarFile.getJarFile(URLJarFile.java:42)\n\tat sun.net.www.protocol.jar.JarFileFactory.get(JarFileFactory.java:78)\n\tat sun.net.www.protocol.jar.JarURLConnection.connect(JarURLConnection.java:85)\n\tat sun.net.www.protocol.jar.JarURLConnection.getJarFile(JarURLConnection.java:69)\n\tat org.apache.catalina.startup.ContextConfig.tldScanJar(ContextConfig.java:906)\n\tat org.apache.catalina.startup.ContextConfig.tldScan(ContextConfig.java:868)\n\tat org.apache.catalina.startup.ContextConfig.start(ContextConfig.java:647)\n\tat org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:243)\n\tat\norg.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:166)\n\tat org.apache.catalina.core.StandardContext.start(StandardContext.java:3582)\n\tat org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1188)\n\tat org.apache.catalina.core.StandardHost.start(StandardHost.java:754)\n\tat org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1188)\n\tat org.apache.catalina.core.StandardEngine.start(StandardEngine.java:363)\n\tat org.apache.catalina.core.StandardService.start(StandardService.java:497)\n\tat org.apache.catalina.core.StandardServer.start(StandardServer.java:2190)\n\tat org.apache.catalina.startup.Catalina.start(Catalina.java:512)\n\tat org.apache.catalina.startup.Catalina.execute(Catalina.java:400)\n\tat org.apache.catalina.startup.Catalina.process(Catalina.java:180)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat\nsun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat java.lang.reflect.Method.invoke(Method.java:324)\n\tat org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:203)\n",
            "date": "20040326T17:04:32",
            "id": 0
        },
        {
            "author": null,
            "body": "The TLD scanning was completely re-written for Tomcat 7 so the error messages there should be a lot better.\n\nI have added the URL (where it isn't null) to the Exception message so it is available for any other code paths that may trigger it. This change has been made to 7.0.x and will be included in 7.0.9 onwards.\n\nI have also proposed the same change for 6.0.x and 5.5.x.\n\nSince 5.5.x is the oldest currently supported version, I am moving this bug to that version.",
            "date": "20110302T08:30:41",
            "id": 1
        },
        {
            "author": null,
            "body": "Fixed in 6.0.x and will be included in 6.0.33 onwards.",
            "date": "20110401T11:12:27",
            "id": 2
        },
        {
            "author": null,
            "body": "Fixed in 5.5.x, will be included in 5.5.34 onwards.",
            "date": "20110410T06:46:35",
            "id": 3
        }
    ],
    "component": "Catalina",
    "description": "org.apache.naming.resources.DirContextURLConnection raises FileNotFoundException\non lines 311, 344, 382 and 396.  It would be more helpful if these provided a\nmessage string.\n\nFor example, change line 344 to\n        throw new FileNotFoundException(getURL().getFile());\n\nThis would make it easier to diagnose mistakes in server configurations.",
    "hasPatch": false,
    "hasScreenshot": false,
    "id": "27988",
    "issuetypeClassified": "IMPROVEMENT",
    "issuetypeTracker": "RFE",
    "priority": "P3 enhancement",
    "product": "Tomcat 5",
    "project": "TOMCAT",
    "summary": "Unhelpful FileNotFoundException raised",
    "systemSpecification": false,
    "version": "5.5.33"
}